lib/rules/no-unneeded-ternary.js

```diff
@@ -134,7 +134,17 @@ module.exports = {
                         node,
                         loc: node.consequent.loc.start,
                         message: "Unnecessary use of conditional expression for default assignment.",
-                        fix: fixer => fixer.replaceText(node, `${astUtils.getParenthesisedText(sourceCode, node.test)} || ${astUtils.getParenthesisedText(sourceCode, node.alternate)}`)
+                        fix: fixer => {
```

Comment:
That doesn't work for some elements. I also thought about using `sourceCode.getText`. But some testcases will fail, as they already have parentheses included. The only possible thing I made, to avoid reassignes is following:

```js
const nodeAlternateText = node.alternate.type === "ConditionalExpression" &&
    !astUtils.isParenthesised(sourceCode, node.alternate) ?
    `(${astUtils.getParenthesisedText(sourceCode, node.alternate)})` :
    astUtils.getParenthesisedText(sourceCode, node.alternate);
```

But it is definitely harder to read.